internal/verifier/sigstore/container/container.go

```diff
@@ -173,43 +160,41 @@ func bundleFromGHAttenstationEndpoint(
 		return fmt.Errorf("error getting attestation reply: %w", err)
 	}
 
-	for i := range attestationReply.Attestations {
-		// FIXME: handle multiple attestations properly
-		att := attestationReply.Attestations[i]
-		protobufBundle, err := unmarhsalAttestationReply(&att)
-		if err != nil {
-			logger.Err(err).Msg("error unmarshaling attestation reply")
-			continue
-		}
+	if len(attestationReply.Attestations) == 0 {
+		return nil
+	}
+	att := attestationReply.Attestations[0]
 
-		cs, err := getCerfificateSummaryFromBundle(protobufBundle)
-		if err != nil {
-			logger.Err(err).Msg("error getting certificate summary from bundle")
-			continue
-		}
+	protobufBundle, err := unmarhsalAttestationReply(&att)
+	if err != nil {
+		logger.Err(err).Msg("error unmarshalling attestation reply")
+		return err
+	}
 
-		digest, err := getDigestFromVersion(version)
-		if err != nil {
-			logger.Err(err).Msg("error getting digest from version")
-			continue
-		}
+	cs, err := getCerfificateSummaryFromBundle(protobufBundle)
+	if err != nil {
+		logger.Err(err).Msg("error getting certificate summary from bundle")
+		return err
+	}
 
-		certID, err := verify.NewShortCertificateIdentity(cs.Issuer, "", "", cs.BuildSignerURI)
-		if err != nil {
-			logger.Err(err).Msg("error getting certificate identity")
-			continue
-		}
+	digest, err := getDigestFromVersion(version)
+	if err != nil {
+		logger.Err(err).Msg("error getting digest from version")
+		return err
+	}
 
-		params.si.IsSigned = true
-		params.si.CertIdentity = &cs.BuildSignerURI
-		params.si.CertIssuer = &cs.Issuer
-		params.digest.algo = containerdigest.Canonical.String()
-		params.digest.bytes = digest
-		params.certID = &certID
-		params.bundle = protobufBundle
-		break
+	certID, err := verify.NewShortCertificateIdentity(cs.Issuer, "", "", cs.BuildSignerURI)
+	if err != nil {
+		logger.Err(err).Msg("error getting certificate identity")
+		return err
```

Comment:
This was a `continue` and is now a `return err`.  I'm assuming this is intended?

(I also notice that we're logging the errors here and then also returning them where they may be logged at the caller.  We might want to consider using the `return fmt.Errorf("error getting certificate identity: %w", err)` form.

Reply:
In particular, it looks like the previous code would try _all_ the attestations until it found one it could verify, while the new code checks that the first attestation can be verified (and then doesn't check the others).

I can certainly be convinced that the new behavior is more correct, but I'm not sure I understand why the change.

Reply:
agreed about the err wrapping.

So this differs between the OCI image where there can only be one and the attestations endpoint where there can be multiple attestations returned. 

After having discussed this with @rdimitrov I think you are right and not only should we return the first that verifies, but we should just return all results and let the upper layer make the decision.

Reply:
My point about the wrapping vs logging is that you can end up with logs that look like:
```
something went wrong!
Unable to frob the foosticator: something went wrong!
User request failed with 502: Unable to frob the foosticator: something went wrong!
```

I'm not sure if that's happening here, but general style would be _either_ to return an error _or_ log a message and _not_ return the error.